conclave-common/src/main/kotlin/com/r3/conclave/common/internal/attestation/AttestationJsonTypes.kt

```diff
@@ -523,68 +481,62 @@ data class EnclaveTcb constructor(val isvsvn: Int) {
  * Utility functions for manual parsing of json primitives and producing sensible error messages.
  */
 private fun JsonNode.checkFieldExists(fieldName: String) {
-    check(this.has(fieldName)) { "No such field $fieldName." }
+    require(this.has(fieldName)) { "No such field $fieldName." }
 }
 
 private fun JsonNode.getInt(fieldName: String): Int {
     checkFieldExists(fieldName)
     val field = this.get(fieldName)
-    check(field.isInt) { "Expected $fieldName to be an integer, got ${field.nodeType}." }
+    require(field.isInt) { "Expected $fieldName to be an integer, got ${field.nodeType}." }
     return field.asInt()
 }
 
 private fun JsonNode.getString(fieldName: String): String {
     checkFieldExists(fieldName)
     val field = this.get(fieldName)
-    check(field.isTextual) { "Expected $fieldName to be a string, got ${field.nodeType}." }
+    require(field.isTextual) { "Expected $fieldName to be a string, got ${field.nodeType}." }
     return field.asText()
 }
 
 private fun JsonNode.getHexEncodedBytes(fieldName: String): OpaqueBytes {
     val stringField = getString(fieldName)
-    val errorMessage by lazy { "$fieldName is not a valid hex string, " }
-    check(stringField.length % 2 == 0) { "$errorMessage, number of characters must be even." }
     return try {
-        val bytes = stringField.chunked(2)
-                .map { it.toInt(16).toByte() }
-                .toByteArray()
-        OpaqueBytes(bytes)
+        OpaqueBytes.parse(stringField)
     } catch (e: NumberFormatException) {
```

Comment:
This doesn't throw a NumberFormatException anymore, and the message for the IllegalArgumentException that's thrown is already sufficient I think.